src/python/pants/core/util_rules/environments.py

```diff
@@ -577,35 +581,40 @@ class OptionField(field_type, EnvironmentSensitiveOptionFieldMixin):  # type: ig
     ]
 
 
-def get_option(name: str, subsystem: Subsystem, env_tgt: EnvironmentTarget):
+def resolve_environment_sensitive_option(name: str, subsystem: Subsystem.EnvironmentAware):
     """Get the option from the `EnvionmentTarget`, if specified there, else from the `Subsystem`.
 
     This is slated for quick deprecation once we can construct `Subsystems` per environment.
     """
 
+    env_tgt = subsystem.env_tgt
+    container_name = name.lstrip("-").replace("-", "_")
```

Comment:
this is the same incantation as in `parser.py`, it wasn't immediately clear how to refactor that.

Reply:
(found a better option that doesn't need this code at all!)